go-zen-chu/nuc-playbook
# support kind install

## Why

- changed minikube support to kind for learning about k8s components

## What

- supported kind installation
- change dir, file structure
- install golang for kind
  - add ansible to docker group

## QA, Evidence


```
z> amasuda ~/dev/nuc-playbook $ ansible-playbook -i hosts.yml site.yml  (git)-[kind-support]
SSH password:

PLAY [all] *********************************************************************

TASK [Gathering Facts] *********************************************************
[WARNING]: Platform linux on host nuc is using the discovered Python
interpreter at /usr/bin/python, but future installation of another Python
interpreter could change this. See https://docs.ansible.com/ansible/2.9/referen
ce_appendices/interpreter_discovery.html for more information.
ok: [nuc]

TASK [kind : install golang] ***************************************************
ok: [nuc]

TASK [kind : install docker] ***************************************************
ok: [nuc]

TASK [kind : Add ansible user to docker group] *********************************
changed: [nuc]

TASK [kind : install kubectl] **************************************************
ok: [nuc]

TASK [install kind] ************************************************************
changed: [nuc]

TASK [status : check partions info] ********************************************
ok: [nuc]

TASK [status : debug] **********************************************************
ok: [nuc] => {
    "msg": {
        "changed": false,
        "disk": {
            "dev": "/dev/sda",
            "logical_block": 512,
            "model": "ATA Samsung SSD 860",
            "physical_block": 512,
            "size": 488386584.0,
            "table": "gpt",
            "unit": "kib"
        },
        "failed": false,
        "partitions": [
            {
                "begin": 1024.0,
                "end": 499712.0,
                "flags": [
                    "boot",
                    "esp"
                ],
                "fstype": "fat32",
                "name": "/dev/sda1",
                "num": 1,
                "size": 498688.0,
                "unit": "kib"
            },
            {
                "begin": 499712.0,
                "end": 484386816.0,
                "flags": [],
                "fstype": "ext4",
                "name": "/dev/sda2",
                "num": 2,
                "size": 483887104.0,
                "unit": "kib"
            },
            {
                "begin": 484386816.0,
                "end": 488386560.0,
                "flags": [],
                "fstype": "linux-swap(v1)",
                "name": "/dev/sda3",
                "num": 3,
                "size": 3999744.0,
                "unit": "kib"
            }
        ],
        "script": "unit 'KiB' print"
    }
}

PLAY RECAP *********************************************************************
nuc                        : ok=8    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
